File: Python3/프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    
    stack = []
    for i in s:
        if not stack:
            if i == '(':
                stack.append('(')
            elif i == ')':
                return False
        else:
            if i == '(':
                stack.append('(')
            elif i == ')':
                stack.pop()
    
    if len(stack) == 0:
        return answer
    else:
        return False

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>
#include <stack>
using namespace std;

bool solution(string s)
{
    bool answer = true;

    stack<int> stack;
    
    for(int i=0; i<s.size(); i++) {
        // '('면 추가 // ')'면 pop  
        if(s[i] == '(') {
            stack.push('(');
        } else if(s[i] == ')' && stack.size() > 0) {
            stack.pop();
        } else {
            return false;
        }
    }
    
    if(stack.size() == 0) {
        answer = true;
    } else {
        answer = false;
    }

    return answer;
}